buzzmobile/tests/test_utils/launch_tools.py

```diff
@@ -44,13 +51,13 @@ def launcher(func):
         def new_test(self):
             """Wrapper around the user provided test that runs a launch file.
             """
-            launch = ROSLauncher(full_name)
+            os.environ['ROS_MASTER_URI'] = self.rosmaster_uri
+            launch = ROSLauncher(full_name, port=self.port)
```

Comment:
`__init__` never returns anything ;)

the broader question, we want to allow all of the following:

    @with_launch_file(*args)
    @mock_node(*node_args)
    @mock_node(*other_args)
    @mock_node(*more_of_these)
    def test func():
        ...

as well as

    @mock_node(*some_args)
    @mock_node(*diff_args)
    def test_function():
        ...

Note that in the second case with_launch is never called. This might be the case if there was no initialization, but in both cases, we need to initialize a "master" launcher for the test through which all nodes are launched. With `with_launch_file`, we can be sure that it will be the first (and last) thing to be executed wrapping the function, so it can *always* take over the port.

Mock_node on the other hand might sometimes need to do it, but only if there is no existing launcher, in which case it needs to know that it is in charge and clean up after itself (shutdown the launcher).

I can add that in as a comment if you want.